Honor pltime/vltime in interface declarations for received delegations

tickerguy opened the pull request:
Currently rtadvd ignores interface pltime/vltime specifications unless the (static) address range is also included in the config file. This extends the validity of a pltime and/or vltime stanza in the config file for an interface to delegated addresses from an upstream provider.  Makes no change to existing behavior unless those stanzas are present.

tickerguy committed:
Update config.c to honor pltime/vltime in interface declarations for received delegations

Currently rtadvd ignores interface pltime/vltime specifications unless the (static) address range is also included in the config file. This extends the validity of a pltime and/or vltime stanza in the config file for an interface to delegated addresses from an upstream provider.  Makes no change to existing behavior unless those stanzas are present.

spmzt commented on usr.sbin/rtadvd/config.c:
For `val64` ~use int~ and move it here
```
int plen;
int64_t val64;
```
EDIT: int64_t was right.

spmzt commented on usr.sbin/rtadvd/config.c:
- Use tab instead
- ~rename val64 to val~ and declare it with `plen` after for-loop.
- move the default value under a validation condition to overwrite out of range values.
- warn the user if the value was out of range.

```suggestion
	MAYHAVE(val, "vltime", DEF_ADVVALIDLIFETIME);
	if (val < 0 || val > 0xffffffff) {
		syslog(LOG_WARNING,
		    "<%s> vltime (%d) for %s/%d on %s "
		    "is out of range, use default value instead.",
		    __func__, val, ntopbuf, pfx->pfx_prefixlen, ifi->ifi_ifname);
		pfx->pfx_validlifetime = DEF_ADVVALIDLIFETIME;
	} else
		pfx->pfx_validlifetime = val;
	MAYHAVE(val, "pltime", DEF_ADVPREFERREDLIFETIME);
	if (val < 0 || val > 0xffffffff) {
		syslog(LOG_WARNING,
		    "<%s> pltime (%d) for %s/%d on %s "
		    "is out of range, use default value instead.",
		    __func__, val, ntopbuf, pfx->pfx_prefixlen, ifi->ifi_ifname);
		pfx->pfx_preflifetime = DEF_ADVPREFERREDLIFETIME;
	} else
		pfx->pfx_preflifetime = val;
```

EDITED: the val64 part is right.

spmzt commented:
Also, please add the `Signed-off-by` tag, fix your git author name (if you want to be your full name), and squash your commits.

Thank you for your contribution.

@bzfbd Thank you too for CC'ing me.

tickerguy committed:
Update usr.sbin/rtadvd/config.c

Accepted.

Co-authored-by: Pouria Mousavizadeh Tehrani <pouria@freebsd.org>

tickerguy committed:
Update config.c

Per the discussion at https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=288426 my original use of int64_t was in fact required; adjusted the patch.  Tested on 14.4-STABLE against my local checked out repo (my gw runs 14.4 at present) and functions as expected.  Sorry didn't catch that on perusal and thanks for the catch.

spmzt reviewed (approved):
I will remove the trailing space during commit.

spmzt commented:
Automated message from ghpr: Thank you for your submission. This PR has been merged to FreeBSD's `main` branch. These changes will appear shortly on our GitHub mirror.
